lib/srv/ctx.go

```diff
@@ -722,6 +726,15 @@ func (c *ServerContext) ExecCommand() (*ExecCommand, error) {
 		requestType = c.request.Type
 	}
 
+	// This block of code gathers and prepares metadata thats needed for user accounting.
+	hostname := c.srv.GetInfo().GetHostname()
+	remoteAddr := c.ConnectionContext.ServerConn.Conn.RemoteAddr()
+	preparedAddr, err := uacc.PrepareAddr(remoteAddr)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
```

Comment:
handle this in `uacc.Open` internally

Reply:
This was discussed in a previous revision. Once you hit `reexec.go` no more logging exists so it's better to do it here and pass the fully parsed remote IP to the child where it can just be used.